File: LeetCode/2033. Minimum Operations to Make a Uni-Value Grid/solution.py
class Solution:
    def minOperations(self, grid: List[List[int]], b: int) -> int:
        arr = []
        for row in grid:
            for x in row:
                arr.append(x)
        mi = min(arr)
        nums = []
        for x in arr:
            if x > 1 and (x-mi)%b != 0:
                return -1
            nums.append((x-mi)//b)
        nums.sort()
        n = len(nums)
        res = sum(abs(x-nums[n//2]) for x in nums)
        return res

File: LeetCode/2033. Minimum Operations to Make a Uni-Value Grid/solution.cpp
class Solution {
public:
    int minOperations(vector<vector<int>>& grid, int x) {
        int n = grid.size(), m = grid[0].size(), mod = grid[0][0] % x;
        vector<int> v;
        for (auto &row : grid)
            for (int cell : row) {
                if (cell % x != mod)
                    return -1;
                v.emplace_back(cell / x);
            }
        int mid = n * m / 2;
        nth_element(v.begin(), v.begin() + mid, v.end());
        int ans = 0;
        for (int vi : v)
            ans += abs(vi - v[mid]);
        return ans;
    }
};


作者：吴自华
链接：https://leetcode-cn.com/circle/discuss/2ELXyY/view/DxDKIC/
来源：力扣（LeetCode）
著作权归作者所有。商业转载请联系作者获得授权，非商业转载请注明出处。